### PR TITLE
show team list page on manage site

### PR DESCRIPTION
**Pull request by YaningX**
show team list page on manage site

### CODE REVIEW

File: open-hackathon-client/src/client/static/js/manage.nav-data.js [hunk 1]
```diff
@@ -37,6 +37,10 @@ angular.module('oh.pages', [])
         state: "manage.users",
         icon: 'fa-users'
       }, {
+        name: "SETTINGS.TEAMS",
+        state: "manage.teams",
+        icon: 'fa-users'
```
Comment (juniwang) [inline open-hackathon-client/src/client/static/js/manage.nav-data.js]: Can we choose another icon other than `fa-users`

File: open-hackathon-client/src/client/static/partials/manage/teams.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        </tr>
+        </thead>
+        <tbody id="registe_list">
+            <tr ng-repeat="team in data.registerTeams">
+                <td><a ng-href= "/site/{{team.hackathon.name}}/team/{{team.id}}">{{team.name}}</a></td>
+                <td><a ng-href="/user/p_{{team.leader.id}}">{{team.leader.name}}</a></td>
+                <td><p ng-repeat="member in team.members"><a ng-href="/user/p_{{member.user.id}}">{{member.user.name}}</a></p></td>
+                <td>{{team.project_name}}</td>
+                <td><p ng-repeat="work in team.works">{{work.uri}} </p></td>
+                <td><p ng-repeat="score in team.scores">{{score.score}} </p></td>
```
Comment (juniwang) [inline open-hackathon-client/src/client/static/partials/manage/teams.html]: both work and score need more sensible display. An uri or score number is not enough. We'd better talk with Jiayun first

File: open-hackathon-server/src/hackathon/hack/team_manager.py [hunk 3]
```diff
@@ -161,6 +161,61 @@ def get_tema(team):
 
         return [get_tema(x) for x in teams]
 
+    def get_admin_hackathon_team_list(self, hackathon_id, name=None, number=None):
```
Comment (juniwang) [inline open-hackathon-server/src/hackathon/hack/team_manager.py]: Where is the pagination? Suppose the method should accept `page_number`, `page_index` or similar

File: open-hackathon-server/src/hackathon/hack/team_manager.py [hunk 4]
```diff
@@ -161,6 +161,61 @@ def get_tema(team):
...
+        :rtype: list
+        :return: a list of team filter by name and number on selected hackathon
+        """
+        query = Q(hackathon=hackathon_id)
+        if name is not None:
+            query &= Q(name__icontains=name)
+
+        try:
+            teams = Team.objects(query).order_by('name')[:number]
+        except ValidationError:
```
Comment (juniwang) [inline open-hackathon-server/src/hackathon/hack/team_manager.py]: Why a single query may raise a `ValidationError`?

File: open-hackathon-server/src/hackathon/hack/team_manager.py [hunk 5]
```diff
@@ -161,6 +161,61 @@ def get_tema(team):
...
+            return []
+
+        # check whether it's anonymous user or not
+        user = None
```
Comment (juniwang) [inline open-hackathon-server/src/hackathon/hack/team_manager.py]: Do we need differentiate logon user with anonymous user?

File: open-hackathon-server/src/hackathon/views/__init__.py [hunk 6]
```diff
@@ -83,6 +82,7 @@ def init_routes():
     api.add_resource(TeamTemplateResource, "/api/team/template")  # select or unselect template for team
 
     # APIs for admin to manage hackathon and hackathon resources, features and users
+    api.add_resource(AdminHackathonTeamListResource, "/api/admin/team/list")  # list teams of hackathon
```
Comment (juniwang) [inline open-hackathon-server/src/hackathon/views/__init__.py]: `/api/admin/team/list` and `/api/hackathon/team/list` do similar things? What's the difference?

Comment (juniwang) [inline open-hackathon-server/src/hackathon/views/__init__.py]: If you do need add a new API, please don't forget to update wiki page

Comment (YaningX) [inline open-hackathon-server/src/hackathon/views/__init__.py]: `/api/admin/team/list` and `/api/hackathon/team/list` do similar things, the difference is that  `/api/admin/team/list` return more detail information about the team, and `/api/hackathon/team/list` is more secure.

**Comment by juniwang** [comment]
Pleasee resolve the conflicts

**Pull request closed without merging**